src/components/Layout/TableOfContents.jsx

```diff
@@ -1,165 +1,90 @@
-import React from "react"
-import Link from 'gatsby-link'
-import styled from 'styled-components'
-
-
-class TableOfContents extends React.Component {
-  constructor() {
-    super();
-    this.nodeListItemsToRender = [];
-    this.currentLevel = 0;
-  }
-
-  formatChapterTitle(title) {
```

Comment:
This method should **not** be in the scope of the class. If you need such a function, it should be declared outside of the class at the top of the document like so:

```js
import React from "react"
import Link from 'gatsby-link'
import styled from 'styled-components'

const formatChapterTitle = (title) => title.split('_').map(word => word[0].toUpperCase() + word.substring(1)).join(' ')

class TableOfContents extends React.Component {
  constructor() {
    super();
    this.nodeListItemsToRender = [];
    this.currentLevel = 0;
  }

  addSubchapterJSX(node) {
    this.nodeListItemsToRender.push(
      <SubchapterLIContainer key={node}>
        <h5>
          {formatChapterTitle(node)}
        </h5>
      </SubchapterLIContainer>
    );
  }

  addChapterJSX(node) {
    this.nodeListItemsToRender.push(
      <ChapterLIContainer key={node}>
        <h5>
          {formatChapterTitle(node)}
        </h5>
      </ChapterLIContainer>
    )
  }
...
```
You are **not** using `this` in the function. So then do not clutter the classes interface with it. In theory it is also a performance issue because the member method would be created for each instance. In practice V8 will optimize this away, but relying on this is bad practice.

As I propose a better approach for handling the chapter titles, I was able to remove the function altogether.